bytecomp/matching.ml

```diff
@@ -82,11 +82,11 @@ type ctx = {left:pattern list ; right:pattern list}
 let pretty_ctx ctx =
   List.iter
     (fun {left=left ; right=right} ->
-      prerr_string "LEFT:" ;
+      Format.eprintf "LEFT:" ;
       pretty_line Format.err_formatter left ;
-      prerr_string " RIGHT:" ;
+      Format.eprintf " RIGHT:" ;
       pretty_line Format.err_formatter right ;
-      prerr_endline "")
+      Format.eprintf "\n")
```

Comment:
Another minor point: the five line above could be merged into one (or maybe 2?) with 2 `%a` specifier: 
`Format.eprintf "LEFT:%a RIGHT:%a" pretty_line left pretty_line right`

Reply:
Yes, much could be done to clean up the code in matching.
But the changes proposed here have been mostly automatic (except for a few places which I had to fix up manually), and I do not plan to tackle a proper cleanup today, simply to make the debug output readable again :)

Reply:
Nevertheless, I did the change you suggested and another similar one :)